src/views/workflow-history/workflow-history-timeline-load-more/workflow-history-timeline-load-more.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+
+import { Spinner } from 'baseui/spinner';
+import { InView } from 'react-intersection-observer';
+
+import { styled } from './workflow-history-timeline-load-more.styles';
+import { type Props } from './workflow-history-timeline-load-more.types';
+
+export default function WorkflowHistoryTimelineLoadMore(props: Props) {
+  if (props.isFetchingNextPage) {
+    return (
+      <styled.SpinnerContainer>
+        <Spinner data-testid="loading-spinner" />
+      </styled.SpinnerContainer>
+    );
+  }
+
+  if (props.error) {
+    return (
+      <styled.EndMessageContainer $isError={true}>
+        Failed to load more items.{' '}
+        <styled.RetryLink
+          onClick={() => props.fetchNextPage()}
+          target="_blank"
+          rel="noreferrer"
+        >
+          Retry manually
+        </styled.RetryLink>
+      </styled.EndMessageContainer>
+    );
+  }
+
+  if (props.hasNextPage) {
+    return (
+      <InView
+        as="div"
+        data-testid="infinite-scroll-spinner"
+        onChange={(inView) => {
+          if (inView && !props.isFetchingNextPage) {
+            // For testing purposes
+            // TODO @assem.hafez: Remove setimeout on page finalization
+            setTimeout(() => props.fetchNextPage(), 2000);
+          }
+        }}
+      >
+        <styled.SpinnerContainer>
+          <Spinner data-testid="loading-spinner" />
```

Comment:
You only need a testid for the spinner or for the InView component.

If you're going with the spinner, you can have a different testid for this spinner to differentiate it from the loading spinner in tests. What I did for the domain workflows table was to keep the testid in "infinite-scroll-spinner".

Reply:
You'll need to keep the data-testid for InView so that mock-intersection-observer intersects with it. You can rename it to something better fi you like

Reply:
change the InView test-Id